src/utilities/bcl/RemoteBCL.cpp

```diff
@@ -290,13 +294,19 @@ void RemoteBCL::updateComponents() {
   }
 
   for (const BCLSearchResult& component : m_componentsWithUpdates) {
-    downloadMeasure(component.uid());
+    downloadComponent(component.uid());
     boost::optional<BCLComponent> newComponent = waitForComponentDownload();
 
     if (newComponent) {
-      boost::optional<BCLComponent> oldComponent = LocalBCL::instance().getComponent(newComponent->uid());
-      if (oldComponent && oldComponent->versionId() != newComponent->versionId()) {
-        LocalBCL::instance().removeComponent(*oldComponent);
+      // delete outdated versions of this component
+      auto it = m_outdatedComponentVersionIdMap.find(component.uid());
```

Comment:
If desired, could just loop over all components here looking for the same uid and different version ids, that would save the need for the new member variables

Reply:
I think I'd prefer that. I've used a different implementation in #5129, can you review that please?